### PR TITLE
Display 404 when going to non-existent market details

### PR DESCRIPTION
**Pull request by yivlad**

### CODE REVIEW

File: packages/app/src/features/errors/ErrorBoundary.tsx [hunk 1]
```diff
@@ -29,6 +29,14 @@ export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundarySt
 
   render() {
```
Comment (oskarvu) [inline packages/app/src/features/errors/ErrorBoundary.tsx:30]: nit, optional: no nested ifs and grouped falback case.


```suggestion
render() {
  const { hasError } = this.state;
  const { children, fallback } = this.props;

  if (!hasError) {
    return children;
  }

  if (!fallback) {
    throw this.state.error;
  }

  return typeof fallback === 'function'
    ? fallback({ error: this.state.error })
    : fallback;
}
```

File: packages/app/src/features/errors/RouterErrorFallback.tsx [hunk 2]
```diff
@@ -7,12 +7,21 @@ import { NotConnected } from './NotConnected'
...
   const error = useRouteError()
+
+  return <ErrorFallback error={error} fullScreen={fullScreen} />
+}
+
+export interface ErrorFallbackProps {
+  fullScreen?: boolean
+  error?: any
+}
+export function ErrorFallback({ fullScreen, error }: ErrorFallbackProps) {
```
Comment (oskarvu) [inline packages/app/src/features/errors/RouterErrorFallback.tsx:24]: Maybe worth to put in a separate file if it's reused.

**Review by oskarvu — APPROVED** [review]

**Review by krzkaczor — APPROVED** [review]

File: packages/app/src/features/errors/ErrorBoundary.tsx [hunk 3]
```diff
@@ -28,10 +28,18 @@ export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundarySt
...
-    if (this.state.hasError) {
-      return this.props.fallback
+    if (!this.state.hasError) {
+      return this.props.children
     }
 
-    return this.props.children
+    if (!this.props.fallback) {
```
Comment (krzkaczor) [inline packages/app/src/features/errors/ErrorBoundary.tsx:35]: this case is not possible according to prop types 🤔

Comment (yivlad) [reply packages/app/src/features/errors/ErrorBoundary.tsx:35]: Right, wondering why linter didn't catch that

**Pull request merged**